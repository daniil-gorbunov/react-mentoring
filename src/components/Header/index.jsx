import React from 'react';
import { Route, Switch } from 'react-router-dom';
import MovieInfo from '../MovieInfo';
import ResultPanel from '../ResultPanel';
import SearchPanel from '../SearchPanel';
import SortPanel from '../SortPanel';
import styles from './style.less';
import moviesService from '../../services/moviesService';

class Header extends React.Component {
  constructor() {
    super();
    this.state = {
      movie: null,
    };
  }

  componentDidMount() {
    moviesService.getMovie()
      .then((movie) => { this.setState({ movie }); });
  }

  render() {
    return (
      <div>
        <div className={styles.header}>
          <Switch>
            <Route path="/film/:filmName">
              <MovieInfo movie={this.state.movie} />
            </Route>
            <Route path="/search/:searchQuery" component={SearchPanel} />
            <Route path="/" component={SearchPanel} />
          </Switch>
        </div>
        <div className={styles.subHeaderContainer}>
          <Switch>
            <Route path="/film/:filmName" component={ResultPanel} />
            <Route path="/search/:searchQuery" component={SortPanel} />
          </Switch>
        </div>
      </div>
    );
  }
}

export default Header;
