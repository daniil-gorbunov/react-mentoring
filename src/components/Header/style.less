.header {
  padding: 10px 50px 20px;
  background: linear-gradient(
    rgba(0, 0, 0, 0.7),
    rgba(0, 0, 0, 0.7)
  ),
  url("/src/assets/images/background.jpg");

  .title-container {
    display: flex;
    justify-content: space-between;

    .title {
      span {
        font-size: 1.2em;
        color: #f55263;
      }
    }

    .search {
      button {
        background-color: #fff;
        color: #f55263;
        font-size: 1em;
      }
    }
  }

  .search-panel-container {
    border: 0;
  }
}

.sub-header-container {
  background-color: #f5f5f5;
}
