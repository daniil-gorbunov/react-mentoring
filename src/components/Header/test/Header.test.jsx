import React from 'react';
import renderer from 'react-test-renderer';
import Header from '..';

jest.mock('react-router-dom', () => ({
  Switch: 'Switch',
  Route: 'Route',
}));
jest.mock('../../MovieInfo', () => 'MovieInfo');
jest.mock('../../SearchPanel', () => 'SearchPanel');
jest.mock('../../SortPanel', () => 'SortPanel');

describe('Header', () => {
  test('renders correctly', () => {
    const header = renderer.create(<Header />);
    expect(header).toMatchSnapshot();
  });
});
