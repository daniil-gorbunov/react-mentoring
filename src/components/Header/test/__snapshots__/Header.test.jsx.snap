// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <Switch>
      <Route
        path="/film/:filmName"
      >
        <Connect(MovieInfoView)
          movie={null}
        />
      </Route>
      <Route
        component={[Function]}
        path="/search/:searchQuery"
      />
      <Route
        component={[Function]}
        path="/"
      />
    </Switch>
  </div>
  <div
    className="subHeaderContainer"
  >
    <Switch>
      <Route
        component={[Function]}
        path="/film/:filmName"
      />
      <Route
        component={[Function]}
        path="/search/:searchQuery"
      />
    </Switch>
  </div>
</div>
`;
